.github/workflows/ci.yml

```diff
@@ -34,32 +34,53 @@ jobs:
         name: Type Check
         runs-on: ubuntu-latest
         steps:
-            - uses: actions/checkout@v2
+            - uses: actions/checkout@v3
             - name: Setup Node
-              uses: actions/setup-node@v1
+              uses: actions/setup-node@v3
               with:
                   node-version: ${{env.NODE_VERSION}}
             - run: npm ci
             - run: npm run typeCheck --if-present
-    build:
-        if: github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v')
+
+    publish-dev:
+        if: github.ref == 'refs/heads/main'
         needs: [lint, typeCheck, test]
         runs-on: ubuntu-latest
         steps:
-            - uses: actions/checkout@v2
-            - uses: actions/setup-node@v1
+            - uses: actions/checkout@v3
+            - uses: actions/setup-node@v3
               with:
                   node-version: ${{env.NODE_VERSION}}
             - run: npm ci
             - run: npm run build
+            - run: mkdir dev-build
+            - run: |
+                cp -r es dev-build | 
+                cp -r lib dev-build | 
+                cp -r dist dev-build |
+                cp -r style dev-build |
+                cp -r type-declarations dev-build |
+                cp README.md dev-build |
+                cp package.json dev-build
+            - uses: actions/upload-artifact@v3
```

Comment:
is upload-artifact being used by the next step (push-to-branch-action)?  I can't tell why this step is needed

Reply:
no it's not. I can delete this, I was using to confirm the directory was correct before I added the push to gh step